### DISCUSSION TITLE
Cannot declare a CTE inside a Recursive CTE

### DISCUSSION THREAD

**Discussion by onderkalaci** [duckdb/duckdb#12256] [General]
### What happens?

According to [cte-syntax](https://duckdb.org/docs/sql/query_syntax/with.html#common-table-expressions), I expect a recursive CTE to allow declaring CTEs inside.

However, it looks like there is an issue.

### To Reproduce

The query that I use below doesn't make a lot of sense, but it is simple, so I prefer to stick to that.

See that `cte_2` is declared inside `cte_1` and used. The same query works fine on Postgres.

```sql
WITH RECURSIVE cte_1 AS (
     with cte_2 as (SELECT * FROM (VALUES (1)) AS t(col))
     SELECT * FROM cte_2 UNION ALL SELECT * FROM (VALUES (1)) AS t(col)
  ) SELECT * FROM cte_1;
```
```text
Catalog Error: Table with name cte_2 does not exist!
Did you mean "temp.information_schema.tables"?
LINE 5:    SELECT * FROM cte_2 UNION ALL SELECT * FROM (VALUES (1)) AS t(col)
) SELECT * FROM cte_1;...
```

Also, if I change the CTE to a subquery, it works:
```sql
WITH RECURSIVE cte_1 AS (
     SELECT * FROM (SELECT * FROM (VALUES (1)) AS t(col)) as cte_2 UNION ALL SELECT * FROM (VALUES (1)) AS t(col)
  ) SELECT * FROM cte_1;
```
```
┌───────┐
│  col  │
│ int32 │
├───────┤
│     1 │
│     1 │
└───────┘
```

Or, if I don't use `RECURSIVE` cte, it works:
```sql
WITH cte_1 AS (
     with cte_2 as (SELECT * FROM (VALUES (1)) AS t(col))
     SELECT * FROM cte_2 UNION ALL SELECT * FROM (VALUES (1)) AS t(col)
  ) SELECT * FROM cte_1;
```
```text
┌───────┐
│  col  │
│ int32 │
├───────┤
│     1 │
│     1 │
└───────┘
```

### OS:

Both MacOs, and Ubuntu 22.04.4 LTS on WSL2

### DuckDB Version:

0.10.3

### DuckDB Client:

CLI

### Full Name:

Onder KALACI

### Affiliation:

  CrunchyData

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have not tested with any build

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by hannes** [comment]
Hey, thanks for filing this but it is more a feature request than a bug in the existing implementation although the error message could be more clear. Moving this to a discussion.

**Comment by rpbouman** [comment]
this query works in v.1.5.5 (possibly earlier):

```sql
WITH RECURSIVE cte_1 AS (
             with cte_2 as (SELECT * FROM (VALUES (1)) AS t(col))
             SELECT * FROM cte_2 UNION ALL SELECT * FROM (VALUES (1)) AS t(col)
          ) SELECT * FROM cte_1;
          
┌───────┐
│  col  │
│ int32 │
├───────┤
│     1 │
│     1 │
└───────┘          
```

(I think this can be marked answered now)
